mr/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandlerBaseTest.java

```diff
@@ -136,18 +136,50 @@ public void after() throws Exception {
     }
   }
 
+  // PARQUET
+
+  @Test
+  public void testScanEmptyTableParquet() throws IOException {
```

Comment:
Rather than creating more test cases, we usually parameterize the test suite. [`TestIdentityPartitionData`](https://github.com/apache/iceberg/blob/425b10cea34eef11cca9cf0d237e02274f6dc958/spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java) is an example. Could you use that approach instead? Or is there a reason why you chose to do it this way?

Reply:
Hi @rdblue, thanks for taking a look! The reason I did that is because these StorageHandler tests already use a different test runner (`StandaloneHiveRunner`), therefore I couldn't use `Parameterized.class` as well. I've tried to outline this rationale in the PR description, but I'm afraid it must have gotten lost in my rather long-winded sentences... :)

Reply:
Sorry, my mistake!

Reply:
I'm afraid this isn't possible with HiveRunner using JUnit4 rules but it is possible using JUnit5 extensions as this doesn't require a different extension to be used, the @ParameterizedTest annotation can go at the method level and then the two play nicely together. So, if/when Iceberg moves to JUnit5 we could refactor all this.